apis/v1alpha1/httproute_types.go

```diff
@@ -327,6 +332,47 @@ type GatewayObjectReference struct {
 	Name string `json:"name" protobuf:"bytes,2,opt,name=name"`
 }
 
+// RouteGatewayStatus describes the status of a route with respect to an
+// associated Gateway.
+type RouteGatewayStatus struct {
+	// Gateway is a reference to a Gateway object that is associated with
+	// the route.
+	Gateway GatewayObjectReference `json:"gateways" protobuf:"bytes,4,rep,name=gateways"`
+	// Conditions describes the status of the route with respect to the
+	// Gateway.  For example, the "Admitted" condition indicates whether the
+	// route has been admitted or rejected by the Gateway, and why.  Note
+	// that the route's availability is also subject to the Gateway's own
+	// status conditions and listener status, which see.
+	Conditions []RouteCondition `json:"conditions,omitempty" protobuf:"bytes,3,rep,name=conditions"`
+}
+
+// RouteConditionType is a type of condition for a route.
+type RouteConditionType string
+
+const (
+	// ConditionRouteAdmitted indicates whether the route has been admitted
+	// or rejected by a Gateway, and why.
+	ConditionRouteAdmitted RouteConditionType = "Admitted"
```

Comment:
I like the idea of this, but wanted to point out that this is a positive polarity condition, whereas all our other conditions are negative polarity. Not necessarily a bad thing, but we may wish to revisit after feedback.

Reply:
Miciah, did you consider using the "Ready" condition for this?

Reply:
At this point, I think we should just do what is natural, given the conversation in upstream. It is alpha and these can be fixed up in a straightforward manner.

Reply:
"Ready" could be misleading:  The controller may have admitted the route but have some unrelated configuration error that prevents the route from being accessible, or the controller may need additional time to make the route accessible (for example, restarting the proxy to reload the configuration).